client/state/data-layer/wpcom/gravatar-upload/index.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * Internal dependencies
+ */
+import {
+	GRAVATAR_UPLOAD_RECEIVE,
+	GRAVATAR_UPLOAD_REQUEST,
+	GRAVATAR_UPLOAD_REQUEST_SUCCESS,
+	GRAVATAR_UPLOAD_REQUEST_FAILURE,
+} from 'state/action-types';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import {
+	bumpStat,
+	composeAnalytics,
+	recordTracksEvent,
+	withAnalytics,
+} from 'state/analytics/actions';
+
+function uploadGravatar( { dispatch }, action ) {
+	const { email, file } = action;
+	dispatch( http( {
+		method: 'POST',
+		path: '/gravatar-upload',
+		body: {},
+		apiNamespace: 'wpcom/v2',
+		formData: [
+			[ 'account', email ],
+			[ 'filedata', file ],
+		],
+	}, action ) );
+}
+
+function announceSuccess( { dispatch }, { file } ) {
+	const fileReader = new FileReader( file );
```

Comment:
this is IE10+ only, is that okay? should we be guarding against it? I thought we supported IE9 but I can't remember